core/db/__init__.py

```diff
@@ -60,7 +60,7 @@ def __init__(self, connection_string=None):
                                     pool_timeout=60)
         self.session_maker = sessionmaker(bind=self.engine,
                                           autocommit=False,
-                                          autoflush=False,
+                                          autoflush=True,
```

Comment:
```
autoflush – When True, all query operations will issue a flush() call to this Session before proceeding. This is a convenience feature so that flush() need not be called repeatedly in order for database queries to retrieve results. It’s typical that autoflush is used in conjunction with autocommit=False. In this scenario, explicit calls to flush() are rarely needed; you usually only need to call commit() (which flushes) to finalize changes.
```